File: src/routing/Catchment.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace DVAN.Routing
{
    public class Catchment : ICatchment
    {
        public List<int>[] sources { get; set; }

        public Catchment(List<int>[] sources)
        {
            this.sources = sources;
        }

        public IEnumerable<int> getNeighbours(int destination)
        {
            var agg = this.sources[destination];
            if (agg == null) {
                return Enumerable.Empty<int>();
            }
            return agg;
        }
    }
}

File: src/main/java/org/tud/oas/routing/Catchment.java
package org.tud.oas.routing;

import java.util.Collections;
import java.util.Iterator;
import java.util.List;

public class Catchment implements ICatchment {
    public List<Integer>[] sources;

    public Catchment(List<Integer>[] sources) {
        this.sources = sources;
    }

    public Iterable<Integer> getNeighbours(int destination) {
        List<Integer> agg = this.sources[destination];
        if (agg == null) {
            return Collections.emptyList();
        }
        return agg;
    }
}